src/main/java/org/opensearch/flowframework/template/TemplateParser.java

```diff
@@ -0,0 +1,164 @@
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for parsing templates.
+ */
+public class TemplateParser {
+
+    private static final Logger logger = LogManager.getLogger(TemplateParser.class);
+
+    // Field names in the JSON. Package private for tests.
+    static final String WORKFLOW = "sequence";
+    static final String NODES = "nodes";
+    static final String NODE_ID = "id";
+    static final String EDGES = "edges";
+    static final String SOURCE = "source";
+    static final String DESTINATION = "dest";
+
+    /**
+     * Prevent instantiating this class.
+     */
+    private TemplateParser() {}
+
+    /**
+     * Parse a JSON representation of nodes and edges into a topologically sorted list of process nodes.
+     * @param json A string containing a JSON representation of nodes and edges
+     * @param workflowSteps A map linking JSON node names to Java objects implementing {@link WorkflowStep}
+     * @return A list of Process Nodes sorted topologically.  All predecessors of any node will occur prior to it in the list.
+     */
+    public static List<ProcessNode> parseJsonGraphToSequence(String json, Map<String, WorkflowStep> workflowSteps) {
+        Gson gson = new Gson();
+        JsonObject jsonObject = gson.fromJson(json, JsonObject.class);
+
+        JsonObject graph = jsonObject.getAsJsonObject(WORKFLOW);
+
+        List<ProcessNode> nodes = new ArrayList<>();
+        List<ProcessSequenceEdge> edges = new ArrayList<>();
+
+        for (JsonElement nodeJson : graph.getAsJsonArray(NODES)) {
+            JsonObject nodeObject = nodeJson.getAsJsonObject();
+            String nodeId = nodeObject.get(NODE_ID).getAsString();
+            // The below steps will be replaced by a generator class that instantiates a WorkflowStep
+            // based on user_input data from the template.
+            WorkflowStep workflowStep = workflowSteps.get(nodeId);
+            // temporary demo POC of getting from a request to input data
+            // this will be refactored into something pulling from user template
+            WorkflowData inputData = WorkflowData.EMPTY;
+            if (List.of("create_index", "create_another_index").contains(nodeId)) {
+                CreateIndexRequest request = new CreateIndexRequest(nodeObject.get("index_name").getAsString());
+                inputData = new WorkflowData() {
+
+                    @Override
+                    public Map<String, Object> getContent() {
+                        // See CreateIndexRequest ParseFields for source of content keys needed
+                        return Map.of("mappings", request.mappings(), "settings", request.settings(), "aliases", request.aliases());
+                    }
+
+                    @Override
+                    public Map<String, String> getParams() {
+                        // See RestCreateIndexAction for source of param keys needed
+                        return Map.of("index", request.index());
```

Comment:
Why are we having `index` in `getParams()` as it will be also provided in content?

Reply:
That's where it is in a CreateIndexRequest that this is a temporary placeholder for

Reply:
I meant for our execution API it will be passed in content and not params.

Reply:
I leave it to you to handle this in #38 however you want to.  The above is placeholder code that will be removed